module/VuFind/src/VuFind/Controller/Plugin/Captcha.php

```diff
@@ -127,7 +128,7 @@ public function verify(): bool
             }
         }
 
-        if (!$captchaPassed && $this->errorMode != 'none') {
+        if (isset($captcha) && !$captchaPassed && $this->errorMode != 'none') {
```

Comment:
This doesn't seem necessary. `active()` should always return false above unless there are captchas, and it can be argued that if this was active for any reason without captchas, the check should fail..

Reply:
That is true, but phpstan can't see it... So I removed `isset` for now and I am going to add `@phpstan-ignore-line`, instead of code change in future PR for phpstan I am working on (final update the level to **1**)

Reply:
Maybe a more elegant solution to this would be to default $message to an empty value at the top of the function, add:

```
if (!$captchaPassed) {
    $message = $captcha->getErrorMessage();
}
```

inside the try..catch block, and check for `!empty($message)` instead of `!$captchaPassed` in the final if statement.

It appears that phpstan is correctly identifying a problem where if the CAPTCHA handler throws an exception, we will try to call a method on a non-object. This refactoring should fix that. It also gives us an opportunity to provide a generic failure message inside the catch block if we want to (which might be a good idea, though I'm not sure exactly what that message should say... it's essentially "CAPTCHA failed due to technical difficulties" or something).

Reply:
(Perhaps you want to make that change in your level 1 PR rather than here, which is why I approved this PR in spite of this one unresolved issue).

Reply:
Yes, I think it would be better to take care of this in next PR